lib/easy.c

```diff
@@ -1056,11 +1056,18 @@ CURLcode curl_easy_pause(struct Curl_easy *data, int action)
     for(i = 0; i < count; i++) {
       /* even if one function returns error, this loops through and frees all
          buffers */
-      if(!result)
-        result = Curl_client_chop_write(data->easy_conn,
+      if(!result) {
+        struct connectdata *conn = data->easy_conn;
+        if(conn && conn->data != data && data->mstate > CURLM_STATE_CONNECT &&
+          data->mstate < CURLM_STATE_COMPLETED)
```

Comment:
@bagder Why is the state check necessary here? I'm not suggesting it's wrong I just don't understand why it would be necessary.

Reply:
Also wouldn't the ownership need to be reverted after the loop completes? since curl_easy_pause can be called from anywhere.

Reply:
@jay I put those checks because I am not 100% sure about `mstate` changes. For me, it is reasonable to check.

For reverting the owner, as far as I've checked, the owner is just changed by multi handler before handling easy handle on demand basic. No assumption can be seen here. But I agree if you say that's better and safer.

Reply:
@jay I'm actually not convinced the state check is necessary, but I also don't think it hurts since this function shouldn't really be called if that state check doesn't match.

Reply:
And yes, I think a reversion at the end of the loop is a good idea so that other parts of the code won't get surprises.

Reply:
Okay, I'll save and restore old owner at the end. Also I confirmed Curl_client_chop_write does not reply on `mstate`, so I will remove `mstate` here.